src/cockpit/beiboot.py

```diff
@@ -181,9 +196,20 @@ def __init__(self, router: Router, destination: str, args: argparse.Namespace):
         super().__init__(router)
 
     async def do_connect_transport(self) -> None:
+        # do we have user/password (Basic auth) from the login page?
+        auth = await self.router.request_authorization("*")
+        ssh_opts = []
+        basic_password = None
+
+        if auth is not None and auth.startswith('Basic '):
+            user, basic_password = base64.b64decode(auth.split(' ', 1)[1]).decode().split(':', 1)
+            if user:  # this can be empty, i.e. auth is just ":"
+                logger.debug("got username %s and password from Basic auth", user)
+                ssh_opts = ['-l', user]
```

Comment:
What happens if you give a username and connect to a destination like `admin@c`.  Who wins?  Should we have a test?

Reply:
The user name wins: `ssh -l admin root@c` logs in as "admin". I did that in commit 38fa110a020c for the shell, happy to do a similar test case for the login page.

Reply:
Correction: We already do have a test, right in this PR:
```py
        # colliding usernames; user names in "Connect to:" are *not* supported,
        # but pin down the behaviour
        try_login("admin", "foobar", server=f"other@{my_ip}")
       check_session()
```

Reply:
Cool!  Didn't see that, thanks!